fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -268,2642 +269,2507 @@ public class ShowExecutor {
-        } else if (stmt instanceof AdminShowReplicaDistributionStmt) {
-            handleAdminShowTabletDistribution();
-        } else if (stmt instanceof AdminShowConfigStmt) {
-            handleAdminShowConfig();
-        } else if (stmt instanceof ShowSmallFilesStmt) {
-            handleShowSmallFiles();
-        } else if (stmt instanceof ShowDynamicPartitionStmt) {
-            handleShowDynamicPartition();
-        } else if (stmt instanceof ShowIndexStmt) {
-            handleShowIndex();
-        } else if (stmt instanceof ShowTransactionStmt) {
-            handleShowTransaction();
-        } else if (stmt instanceof ShowPluginsStmt) {
-            handleShowPlugins();
-        } else if (stmt instanceof ShowSqlBlackListStmt) {
-            handleShowSqlBlackListStmt();
-        } else if (stmt instanceof ShowDataCacheRulesStmt) {
-            handleShowDataCacheRulesStmt();
-        } else if (stmt instanceof ShowAnalyzeJobStmt) {
-            handleShowAnalyzeJob();
-        } else if (stmt instanceof ShowAnalyzeStatusStmt) {
-            handleShowAnalyzeStatus();
-        } else if (stmt instanceof ShowBasicStatsMetaStmt) {
-            handleShowBasicStatsMeta();
-        } else if (stmt instanceof ShowHistogramStatsMetaStmt) {
-            handleShowHistogramStatsMeta();
-        } else if (stmt instanceof ShowResourceGroupStmt) {
-            handleShowResourceGroup();
-        } else if (stmt instanceof ShowUserStmt) {
-            handleShowUser();
-        } else if (stmt instanceof ShowCatalogsStmt) {
-            handleShowCatalogs();
-        } else if (stmt instanceof ShowComputeNodesStmt) {
-            handleShowComputeNodes();
-        } else if (stmt instanceof ShowAuthenticationStmt) {
-            handleShowAuthentication();
-        } else if (stmt instanceof ShowCreateExternalCatalogStmt) {
-            handleShowCreateExternalCatalog();
-        } else if (stmt instanceof ShowCharsetStmt) {
-            handleShowCharset();
-        } else if (stmt instanceof ShowStorageVolumesStmt) {
-            handleShowStorageVolumes();
-        } else if (stmt instanceof DescStorageVolumeStmt) {
-            handleDescStorageVolume();
-        } else if (stmt instanceof ShowPipeStmt) {
-            handleShowPipes();
-        } else if (stmt instanceof DescPipeStmt) {
-            handleDescPipe();
-        } else if (stmt instanceof ShowFailPointStatement) {
-            handleShowFailPoint();
-        } else if (stmt instanceof ShowDictionaryStmt) {
-            handleShowDictionary();
-        } else if (stmt instanceof ShowBackendBlackListStmt) {
-            handleBackendBlackList();
-        } else {
-            handleEmpty();
-        }
-
-        List<List<String>> rows = doPredicate(stmt, stmt.getMetaData(), resultSet.getResultRows());
-        return new ShowResultSet(resultSet.getMetaData(), rows);
+    public ShowResultSet execute(ShowStmt statement, ConnectContext context) {
```

Comment:
looks like this method is good to be `static`, so there is no need `new ShowExecutor().execute()`, just `ShowExecutor.execute()`

Reply:
Static ones are not easy to dependency injection. Subsequent PRs will use ShowExecutor as a globally unique object.